odl/util/ufuncs.py

```diff
@@ -160,68 +136,24 @@ def __init__(self, vector):
         """Create ufunc wrapper for vector."""
         self.vector = vector
 
+    # Reductions for backwards compatibility
+    def sum(self):
+        return self.vector.sum()
 
-# Add ufunc methods to ufunc class
-for name, n_in, n_out, doc in UFUNCS:
-    method = wrap_ufunc_base(name, n_in, n_out, doc)
-    setattr(NtuplesBaseUfuncs, name, method)
-
-# Add reduction methods to ufunc class
-for name, doc in REDUCTIONS:
-    method = wrap_reduction_base(name, doc)
-    setattr(NtuplesBaseUfuncs, name, method)
-
-
-# Optimized implementation of ufuncs since we can use the out parameter
-# as well as the data parameter to avoid one call to asarray() when using an
-# NumpyNtuplesVector
-def wrap_ufunc_ntuples(name, n_in, n_out, doc):
-    """Add ufunc methods to `NumpyNtuplesUfuncs`."""
-
-    # Get method from numpy
-    wrapped = getattr(np, name)
-    if n_in == 1:
-        if n_out == 0:
-            def wrapper(self):
-                return wrapped(self.vector)
-
-        elif n_out == 1:
-            def wrapper(self, out=None):
-                if out is None:
-                    out = self.vector.space.element()
-                wrapped(self.vector, out.data)
-                return out
-
-        elif n_out == 2:
-            def wrapper(self, out1=None, out2=None):
-                if out1 is None:
-                    out1 = self.vector.space.element()
-                if out2 is None:
-                    out2 = self.vector.space.element()
-
-                y1, y2 = wrapped(self.vector, out1.data, out2.data)
-                return out1, out2
-
-        else:
-            raise NotImplementedError
+    def prod(self):
+        return self.vector.prod()
 
-    elif n_in == 2:
-        if n_out == 1:
-            def wrapper(self, x2, out=None):
-                if out is None:
-                    out = self.vector.space.element()
+    def min(self):
+        return self.vector.min()
```

Comment:
I don't think so but I'll check. The point is that `__array_ufunc__` is always *available*, no matter which Numpy version is used. So the methods using it will always find it. The only difference is the behavior of `numpy.some_ufunc(odl_vec, ...)` which uses the new interface from 1.13 on, and doesn't for older versions.